c_src/ep_encoder.c

```diff
@@ -686,7 +684,28 @@ get_oneof_field(ErlNifEnv *env, ERL_NIF_TERM term, ep_node_t *node, ERL_NIF_TERM
         return NULL;
     }
 
-    return bsearch((&(array[0])), node->fields, node->size, sizeof(ep_field_t), get_field_compare_sub_name);
+    // We try and fetch the specific field that was set as part of the oneof field.
+    // oneof field values are set directly to the actual record if the value is a new message.
+    // If the value is one of the basic types - then it is a tuple with the variable name followed by the value.
+    // Here, we first check if the value is another message by checking if the prefix is set to pb_.
+    // If not then we expect the value to be one of the basic types and compare the variable name.
+
+    if(strncmp(buf, "pb_", 3) == 0) {
+        *out = term;
+        _free(buf);
+        return bsearch((&(array[0])), node->fields, node->size, sizeof(ep_field_t), get_field_compare_sub_name);
+    } else {
+        *out = array[1];
+        _free(buf);
+        ep_field_t *result = node->fields;
+        for (int i = 0; i < node->size; ++i) {
+            result = node->fields + i*sizeof(ep_field_t);
```

Comment:
You can get rid of line 700.
```suggestion
            ep_field_t *result = node->fields + i*sizeof(ep_field_t);
```